integration-tests/src/test/java/oracle/weblogic/kubernetes/ItRetryImprovements.java

```diff
@@ -0,0 +1,283 @@
+    String opServiceAccount = opNamespace + "-sa";
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, opServiceAccount, false, 0, domainNamespace);
+  }
+
+  // This method is needed in this test class to delete uncompleted domain to restore the env
+  @AfterEach
+  public void tearDown() {
+    deleteDomainResource(domainNamespace, domainUid);
+  }
+
+  /**
+   * Create a domain-in-image domain before the secret for admin credentials is created,
+   * the domain should fail to start with SEVERE error and the Operator should start retrying
+   * in a specified failure.retry.interval.seconds and failure.retry.limit.minutes.
+   * Also log a clear message in the domain Failed condition with the cause,
+   * actions to fix the problem, and indicates that the next retry time and when the retry will stop.
+   */
+  @Test
+  @DisplayName("Create a domain without WLS secret. Verify that retry occurs and handles SEVERE error as designed.")
+  void testRetryOccursAsExpectedAndThrowSevereFailures() {
+    // verify that the operator starts retrying in the intervals specified in domain.spec.failureRetryIntervalSeconds
+    // when a SEVERE error occurs and clear message is logged.
+    Long failureRetryLimitMinutes = Long.valueOf("1");
+    createDomainWithoutWlsSecret(failureRetryLimitMinutes);
+
+    String retryOccurRegex = new StringBuffer(".*WebLogicCredentials.*\\s*secret.*")
+        .append(wlSecretName)
+        .append(".*not\\s*found\\s*in\\s*namespace\\s*.*")
+        .append(domainNamespace)
+        .append(".*Will\\s*retry\\s*next\\s*at.*and\\s*approximately\\s*every\\s*")
+        .append(FAILURE_RETRY_INTERVAL_SECONDS)
+        .append("\\s*seconds\\s*afterward\\s*until.*\\s*if\\s*the\\s*failure\\s*is\\s*not\\s*resolved").toString();
+
+    testUntil(() -> findStringInDomainStatusMessage(retryOccurRegex), logger, "retry occurs as expected");
+
+    // verify that the operator stops retrying when the maximum retry time is reached
+    String retryMaxValueRegex = new StringBuffer(".*operator\\s*failed\\s*after\\s*retrying\\s*for\\s*.*")
+        .append(failureRetryLimitMinutes)
+        .append(".*\\s*minutes.*\\s*Please\\s*resolve.*error\\s*and.*update\\s*domain.spec.introspectVersion")
+        .append(".*to\\s*force\\s*another\\s*retry\\s*.*").toString();
+
+    testUntil(() -> findStringInDomainStatusMessage(retryMaxValueRegex),
+        logger, "retry ends as expected after {0} minutes retry", failureRetryLimitMinutes);
+
+    // verify that SEVERE level error message is logged in the Operator log
+    String opLogSevereErrRegex = new StringBuffer(".*SEVERE")
+        .append(".*WebLogicCredentials.*\\s*secret.*")
+        .append(wlSecretName)
+        .append(".*not\\s*found\\s*in\\s*namespace\\s*.*")
+        .append(domainNamespace).toString();
+
+    testUntil(() -> findStringInOperatorLog(opLogSevereErrRegex), logger, "SEVERE error found in Operator log");
+  }
+
+  /**
```

Comment:
Can we merge the two methods into a single method to verify the e2e usecase

Reply:
better not because we need search string in Operator log but search string in Domain config. I can move findStringInOperatorLog to OperatorUtil though to make it a util method